### DISCUSSION TITLE
click events from plotly figures

### DISCUSSION THREAD

**Discussion by tejakondury** [Kitware/trame#322] [General] [answered]
Hi, I am really new to trame and I don't know much about javascript.
I am trying to build an application that requires extracting a click event from one plotly figure and use it to update other figures on a page. I am struggling to understand the approach to do this, do I need to get click events from plotly widgets? or can I somehow do it from state.change I couldn't find much in the docs, can you please help?

**Comment by jourdain** [comment, marked as answer]
By clicking, do you mean selection? If so, you must rely on the plotly JavaScript API. You can see [an example here](https://github.com/Kitware/trame/tree/master/examples/06_vtk/Applications/RemoteSelection).
Then you should use [some of the events available](https://github.com/Kitware/trame-plotly/blob/master/trame_plotly/widgets/plotly.py#L40-L66) on your chart.

**Reply by tejakondury** [reply]
Thank you for your response! I should have been more clear in what I was trying to do. I have a plotly bar chart with some ids and their values and a map which shows the locations of all the ids, as two separate figures in the page. I want to be able to select a single id from the bar chart and I want to update the map to only show its location. Any advice on how to approach this? Thanks for your help!

**Reply by jourdain** [reply]
That match my understanding. You want to perform a selection on the plotly chart (single point), then based on that ID sent to the server, you want to update the content of another chart.

The example provided are doing exactly that.

**Reply by CATdj** [reply]
Hi Jourdain, I tried the example, if I select the points in the vtk window, the plot will update, but if I select in the plot, nothing happens, what could be wrong?

**Reply by jourdain** [reply]
Indeed the last update of trame-plotly has messed up something on the events... But if you use `trame-plotly==2.1.1` it will just work. 
Once a fixed is pushed to trame-plotly, it will be back to normal. Thanks for reporting the issue.

**Reply by CATdj** [reply]
Thank you, it works like a charm!

**Reply by jourdain** [reply]
Fixed in https://github.com/Kitware/trame-plotly/releases/tag/v3.0.2

**Reply by CATdj** [reply]
awesome! I tried to capture click event for a bar chart, but the selected_point_idxs are always 0, I'm not familiar with JavaScript, can you please tell me what changes I need to make? this is the code I'm using: html_fig=plotly.Figure(create_fig, click=(on_chart_selection, "[$event?.points.map(({pointIndex}) => pointIndex)]",))

**Reply by jourdain** [reply]
Please share your full example if you want guidance.

**Reply by CATdj** [reply]
Thank you Jourdain, although the return pointIndex is always 0, but I found the label or x is correct, I'll just use them to identify which bar is clicked. Another question not related to this, how do I highlight a cell based on it's ID similar like you highlight the nodes in the example?

**Reply by jourdain** [reply]
You mean a vtk cell?

**Reply by CATdj** [reply]
yes

**Reply by jourdain** [reply]
https://examples.vtk.org/site/
